ara/api/models.py

```diff
@@ -240,6 +240,27 @@ def __str__(self):
         return "<Host %s:%s>" % (self.id, self.name)
 
 
+class DistinctHost(models.Model):
+    """
+    Distinct list of hosts with latest `Host` object id related.
+
+    We can not inherit from Base because we want to use `name` as primary key.
+    """
+
+    class Meta:
+        db_table = "distinct_hosts"
+
+    name = models.CharField(max_length=255, primary_key=True)
```

Comment:
From an implementation standpoint, I was wondering whether we should use the name as the primary key or keep the usual integer and have a name field (with UNIQUE) like we do for labels for example: https://github.com/ansible-community/ara/blob/31f090e4119d1ac55a9807b9e605082cdd3db1b9/ara/api/models.py#L69

Having the name as a primary key provides the ability to do something like: ``/api/v1/distincthosts/<name>`` (rather than ``<id>``) which is nifty. This avoids needing to search for it like ``/api/v1/distincthosts?name=<name>`` and then recover the ID from the list, though the name search is partial so it could potentially return more than one result if the name search is a subset of another host name.

More importantly, it doesn't prevent us from searching by name anyway -- i.e, https://dev.demo.recordsansible.org/hosts?name=localhost returns localhost in addition to localhost1, localhost2, etc.

With that in mind, it's probably a good idea to keep name as the primary key.